process/sync/metablock.go

```diff
@@ -418,16 +423,28 @@ func (boot *MetaBootstrap) SyncBlock() error {
 	}
 
 	if boot.isForkDetected {
-		log.Info(fmt.Sprintf("fork detected at nonce %d with hash %s\n",
-			boot.forkNonce,
-			core.ToB64(boot.forkHash)))
+		isForcedFork := boot.forkNonce == math.MaxUint64 && boot.forkHash == nil
```

Comment:
I do not really like the direction of the forking mechanism is going. We need to make a new technical analysis document, review it and maybe rewrite some of this code.

Reply:
Agree. We already had a discussion, but we also should put it in a document to be easily reviewed and also analyzed if there could be other possible edge cases.